validations/tests/validations.rs

```diff
@@ -0,0 +1,3000 @@
+    let reveal_signature = sign_t(&reveal_body);
+    let commitment = reveal_signature.signature.hash();
+
+    let commit_transaction = CommitTransaction::new(
+        CommitTransactionBody::new(
+            dr_pointer,
+            commitment,
+            DataRequestEligibilityClaim::default(),
+        ),
+        vec![KeyedSignature {
+            signature: Signature::default(),
+            public_key: public_key.clone(),
+        }],
+    );
+    let reveal_transaction = RevealTransaction::new(reveal_body, vec![reveal_signature]);
+
+    // Include CommitTransaction in DataRequestPool
+    dr_pool
+        .process_commit(&commit_transaction, &fake_block_hash)
+        .unwrap();
+    dr_pool.update_data_request_stages();
+
+    let (h, n, fee) = validate_reveal_transaction(&reveal_transaction, &dr_pool).unwrap();
+    assert_eq!(h, dr_pointer);
+    assert_eq!(n, 5);
+    assert_eq!(fee, 100);
+
+    // Create other reveal
+    let reveal_body2 = RevealTransactionBody::new(
+        dr_pointer,
+        vec![0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
+        public_key.pkh(),
+    );
+    let reveal_signature2 = sign_t(&reveal_body2);
+    let reveal_transaction2 = RevealTransaction::new(reveal_body2, vec![reveal_signature2]);
+
+    let error = validate_reveal_transaction(&reveal_transaction2, &dr_pool);
+    assert_eq!(
+        error.unwrap_err().downcast::<TransactionError>().unwrap(),
+        TransactionError::MismatchedCommitment
+    );
+}
+
+fn dr_pool_with_dr_in_tally_stage(reveal_value: Vec<u8>) -> (DataRequestPool, Hash) {
+    // Create DataRequestPool
+    let mut dr_pool = DataRequestPool::default();
+
+    // Create DRTransaction
+    let fake_block_hash = Hash::SHA256([1; 32]);
+    let epoch = 0;
+    let dr_output = DataRequestOutput {
+        witnesses: 5,
+        reveal_fee: 100,
+        data_request: example_data_request(),
+        ..DataRequestOutput::default()
+    };
+    let dr_transaction = DRTransaction {
+        body: DRTransactionBody::new(vec![], vec![], dr_output),
+        ..DRTransaction::default()
+    };
+    let dr_pointer = dr_transaction.hash();
+
+    // Include DRTransaction in DataRequestPool
+    dr_pool.process_data_request(&dr_transaction, epoch);
+    dr_pool.update_data_request_stages();
+
+    // Hack: get public key by signing an empty transaction
+    let public_key = sign_t(&RevealTransactionBody::default()).public_key.clone();
+
+    // Create Reveal and Commit
+    // Reveal = empty array
+    let reveal_body = RevealTransactionBody::new(dr_pointer, reveal_value, public_key.pkh());
+    let reveal_signature = sign_t(&reveal_body);
+    let commitment = reveal_signature.signature.hash();
+
+    let commit_transaction = CommitTransaction::new(
+        CommitTransactionBody::new(
+            dr_pointer,
+            commitment,
+            DataRequestEligibilityClaim::default(),
+        ),
+        vec![KeyedSignature {
+            signature: Signature::default(),
+            publi
```

Comment:
In the future, a block without any transaction will be invalid. So maybe it would be need to add a TODO. But maybe better in block_validation function